testservices/novaservice/service.go

```diff
@@ -801,27 +880,79 @@ func (n *Nova) addServerFloatingIP(serverId string, ipId string) error {
 	}
 	fips = append(fips, ipId)
 	n.serverIPs[serverId] = fips
+	if err := n.addFloatingIPToServerAddresses(serverId, fip.IP); err != nil {
+		return err
+	}
+	return nil
+}
+
+// addFloatingIPToServerAddresses adds a floating ip address to the servers list
+// of Addresses to facilitate juju openstack provider tests.
+func (n *Nova) addFloatingIPToServerAddresses(serverId, address string) error {
+	server, err := n.server(serverId)
+	if err != nil {
+		return err
+	}
+	newAddresses := server.Addresses["private"]
+	if strings.Contains(address, ":") {
```

Comment:
Could use net.IP to do a more sane determination of 4 vs 6, but doesn't matter much for test code.